inc/winrt/Microsoft.Toolkit.Win32.UI.XamlHost.h

```diff
@@ -0,0 +1,159 @@
+// WARNING: Please don't edit this file. It was generated by C++/WinRT v2.0.190730.2
```

Comment:
generated file, should this be in this change?

Reply:
I see that is the point of this approach, check in the generated code to avoid the .idl file...

I'm concerned about versioning problems with this, is the output of cppwinrt.exe stable enough to commit? are there detect-version-mismatch problems with this?

Reply:
I lean towards the "include a .idl file" over this as it avoids the committing generated code. yes, idls are clunky, but it is an honest representation of the design.

Reply:
the challenge with the IDL approach is that the generated code assumes that the type lives in a DLL, because the activation code that runs when you instantiate the projection type will do the LoadLibrary lookup based on the type name.
See: https://github.com/asklar/xaml-islands/blob/67f3271602947929bced6e41ec80b758eb6c8549/inc/XamlApplication.h#L68-L69
where I replace the activation codepath with calling the factory directly instead of going through the cppwinrt factory code (which relies on locating the dll that hosts the GetActivationFactory for that type based on the type name)

Reply:
not if you call winrt::make<> and even if you don't, with the -opt option passed to cppwinrt.exe activation goes directly to the C++ constructor.

see [my project](https://github.com/ChrisGuzak/Win32XamlApp/blob/master/Minimal/Win32XamlApp.cpp#L64), it works as expected.